### ISSUE TITLE
train the model from scratch

### ISSUE THREAD

**Issue by yo3nglau** [DeweiHu/Medical-Domain-Diffusion#4]
Hi Dewei,

I am trying to train the model from scratch. I preprocessed the training data into a `pickle` style following the guidance of `README.md` and source codes. I found that normalizing the data into the range [0, 1] and then pickling them worked well (please point out if it is wrong). Somehow, (**Q1**) I cannot determine if I should use `template` or not, which conducted histogram matching as far as I know.

I tried them all but did not generate images with a similar style to your samples. This might be that I just used `OCTA_6mm/OCTA(FULL)` of the `OCTA-500` dataset. (**Q2**) So may I ask if the released `diffusion.octa500.pt` was trained with all data of the `OCTA-500` dataset (i.e., both under `3mm` and `6mm`, `OCTA(FULL)`, `OCTA(ILM_OPL)`, `OCTA(OPL_BM)`) at the same time?

Besides, the source codes utilized patch sampling for training with the parameter `p_size` (default 256), (**Q3**) so did it imply the optimal input image shape? Finally, (**Q4**) how could one adjust training parameters (e.g., `num_sample`, `batch_size`, `beta_start`, `beta_end`, `T`, `n_epoch`, `lr`) given more or fewer data?

Thank you for your contribution. I look forward to your early response.

Generation result (256x256):
![10301_gen_256_new](https://github.com/user-attachments/assets/902d1bcc-bbd9-41de-8a11-5f4e097b42b7)

Generation result (256x256) with `template`:
![10301_gen_256_new_template](https://github.com/user-attachments/assets/08f73627-8f8d-4e65-9fee-9b76e65b0ab3)

Missing vessels in generation result (256x256) with `template`:
![10301_gen_256_new_template_marked](https://github.com/user-attachments/assets/21414a58-e307-4cff-be4a-195f0a4401d0)

**Comment by DeweiHu** [comment]
Hi there,

Thank you for trying these out, to address your question:

* **Q1**: In my experiments, sometimes the generated images are not perfectly matched with the target data distribution, so I do the histogram matching with a random sample from the target domain. I think I discarded this part in my most recent application since this does not help for the downstream segmentation task. It will be good if you can use an extra constraint during training to avoid such mismatch in histogram.

* **Q2**:  For the OCTA500 dataset, I only used the projection of shallow retinal layers (ILM_OPL). Thanks for bring this up, I will add a data discription in the readme.

* **Q3**: I use the (256, 256) patches randomly sampled from each image to train the model in order to avoid paddings on the boundary. This has nothing to do with the optimal image shape, you can definitely use it with any image dimension. During inference, you can do a sliding window.

* **Q4**: If you get fewer data (e.g., ROSE dataset), you can sample more patches in an image by increasing num_sample. However, the distribution will not be well defined given insufficient training data. The n_epoch can be decrease if the dataset is huge, the model should converge pretty quickly (in a few epochs) based on my observation.

**Comment by yo3nglau** [comment]
Great! I will remake several experiments based on your suggestions. Thanks for your immediate reply.
